numpy/core/src/umath/loops.c.src

```diff
@@ -2862,6 +2862,43 @@ OBJECT_sign(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED
     Py_XDECREF(zero);
 }
 
+/**begin repeat1
+ * #kind = isnan, isinf, isfinite#
+ * #func = npy_isnan, npy_isinf, npy_isfinite#
+ * #OP = ||, ||, &&#
+ **/
+NPY_NO_EXPORT void
+OBJECT_@kind@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
+{
+    double dbl;
+    Py_complex cplx;
+    int v;
+
+    UNARY_LOOP {
+        PyObject *in1 = *(PyObject **)ip1;
```

Comment:
I am almost 100% sure that I don't need to `Py_INCREF(in1)` here or anywhere else, but I feel like someone should verify that since I do pass it to `PyFloat_AsDouble` and `PyComplex_AsCComplex` down the line.